src/otx/algo/action_classification/openvino_model.py

```diff
@@ -0,0 +1,115 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+"""Custom OpenVINO model wrappers for video recognition."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Callable
+
+import numpy as np
+from openvino.model_api.adapters.utils import RESIZE_TYPES, InputTransform
+from openvino.model_api.models.model import Model
+from openvino.model_api.models.utils import (
+    ClassificationResult,
+)
+
+if TYPE_CHECKING:
+    from openvino.model_api.adapters import OpenvinoAdapter
+
+
+def softmax_numpy(x: np.ndarray) -> np.ndarray:
+    """Softmax numpy."""
+    x = np.exp(x - np.max(x))
+    x /= np.sum(x)
+    return x
+
+
+def get_multiclass_predictions(logits: np.ndarray, activate: bool = False) -> ClassificationResult:
+    """Get multiclass predictions."""
+    index = np.argmax(logits)
+    if activate:
+        logits = softmax_numpy(logits)
+
+    return ClassificationResult([(index, index, logits[index])], np.ndarray(0), np.ndarray(0), np.ndarray(0))
```

Comment:
Why don't you create an output sink for softmax at model exportation?

Reply:
Actually, softmax is already performed by model. That's the reason why default value of activate is False. This is for unexpected situation which model do not calculate softmax on its final output

Reply:
We assume that this class only allows the model exported by OTX. From your explanation, there is no case that the model output is not probability.

Reply:
8a90193 Done